Automattic/wp-calypso
# BackupBrowser: Add file browser UI

## Proposed Changes
* Add file browser UI in backup contents page
* Add icons per file type as specified in p1HpG7-mQf-p2#comment-63627

## Follow-up tasks
The unfinished tasks will be addressed in a new PR (multiple if needed).
- [ ] Review and address `@TODO`'s in the code.
- [x] Prevent trying to fetch contents on directory that don't have children.
- [x] Review design specs: p1HpG7-mQf-p2
- [ ] Find a way to consistently display the odd/even row colors on nested items.
- [ ] Review accessibility
- [ ] Truncate long filenames
- [ ] Horizontal scroll

## Demo
https://github.com/Automattic/wp-calypso/assets/1488641/15f3b616-5359-4712-addb-6dca9b3f803b

## Testing Instructions
* Checkout this branch
* Start your local environment using `yarn start-jetpack-cloud`
* Select one of your sites with a Jetpack VaultPress Backup plan
* Navigate to VaultPress Backup page
* Pick one date with a backup available (use the calendar and click on any date marked with a dot).
* Copy the link from the `Download Backup` button and replace `download` with `contents` and navigate to that page.
  * You might need to pick a backup ID from RWDB and replace the last parameter in the URL with it.
* You should see the file browser and should be able to navigate through it.

## Pre-merge Checklist
